### PR TITLE
Add metacontact to the CCA manager

### LINKED ISSUE #106: Add Meta-Contact to contact
Allow the CCA members to add more personal information that they are willing to share as a form of ice-breaker.

### PR DESCRIPTION
**Pull request by alphajae11**
Closes #106 

Add metacontact to the CCA manager

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AddCommand.java [hunk 1]
```diff
@@ -35,6 +36,7 @@ public class AddCommand extends Command {
             + PREFIX_ADDRESS + "311, Clementi Ave 2, #02-25 "
             + PREFIX_ROLE + "friends "
             + PREFIX_ROLE + "owesMoney"
```
Comment (JuliaPoo) [inline src/main/java/seedu/address/logic/commands/AddCommand.java:38]: Formatting error: `"owesMoney"` should be `"owesMoney "`

File: src/main/java/seedu/address/logic/commands/AddCommand.java [hunk 2]
```diff
@@ -35,6 +36,7 @@ public class AddCommand extends Command {
...
+            + PREFIX_METADATA + "I have food allegeries - Egg and Peanuts"
```
Comment (JuliaPoo) [inline src/main/java/seedu/address/logic/commands/AddCommand.java:39]: Formatting error: `"I have food allegeries - Egg and Peanuts"` should be `"I have food allegeries - Egg and Peanuts "`

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 3]
```diff
@@ -48,6 +50,7 @@ public class EditCommand extends Command {
             + "[" + PREFIX_ADDRESS + "ADDRESS] "
             + "[" + PREFIX_ROLE + "ROLE] "
             + "[" + PREFIX_CCA + "CCA]...\n"
+            + "[" + PREFIX_METADATA + "Meta-data]...\n"
```
Comment (JuliaPoo) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:53]: Formatting error: 
```
+ "[" + PREFIX_CCA + "CCA]...\n"
+ "[" + PREFIX_METADATA + "Meta-data]...\n"
```

should be

```
+ "[" + PREFIX_CCA + "CCA] "
+ "[" + PREFIX_METADATA + "Meta-data]...\n"
```

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 4]
```diff
@@ -160,6 +166,7 @@ public EditPersonDescriptor(EditPersonDescriptor toCopy) {
             setAddress(toCopy.address);
             setRoles(toCopy.roles);
             setCcas(toCopy.ccas);
+            setMetadata(toCopy.metadata);
```
Comment (JuliaPoo) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:169]: Github doesnt allow me to comment below but `isAnyFieldEdited` needs to be editted.

**Review by JuliaPoo — CHANGES_REQUESTED** [review]
I did not look through all of the code yet or ran the branch yet, but in general, be more thorough in understand where needs to be modified.

The reason why this feature seems to have so many code changes is because you did not make metadata an optional. If you did your changes dont have to be so invasive.

File: src/test/data/JsonSerializableAddressBookTest/duplicatePersonAddressBook.json [hunk 5]
```diff
@@ -9,7 +9,8 @@
     "CCAs" : [ "NUS Cycling" ],
     "amount" : {
       "value" : "0.0"
-    }
+    },
+    "Metadata" : "I like to do leetcode everyday."
```
Comment (JuliaPoo) [inline src/test/data/JsonSerializableAddressBookTest/duplicatePersonAddressBook.json:13]: Shoud not be "Metadata", should be "metadata".

**Review by JuliaPoo — CHANGES_REQUESTED** [review]
Still have not looked through everything or ran ur code yet, but i found one more mistake

**Review by JuliaPoo — APPROVED** [review]

File: src/main/java/seedu/address/logic/commands/AddCommand.java [hunk 6]
```diff
@@ -34,7 +35,8 @@ public class AddCommand extends Command {
             + PREFIX_EMAIL + "johnd@example.com "
             + PREFIX_ADDRESS + "311, Clementi Ave 2, #02-25 "
             + PREFIX_ROLE + "friends "
-            + PREFIX_ROLE + "owesMoney"
+            + PREFIX_ROLE + "owesMoney "
+            + PREFIX_METADATA + "I have food allegeries - Egg and Peanuts "
```
Comment (chuahjiajie) [inline src/main/java/seedu/address/logic/commands/AddCommand.java:39]: Is this metadata relevant to our apps vision?

File: src/main/java/seedu/address/logic/parser/CliSyntax.java [hunk 7]
```diff
@@ -13,4 +13,5 @@ public class CliSyntax {
     public static final Prefix PREFIX_ROLE = new Prefix("r/");
     public static final Prefix PREFIX_CCA = new Prefix("c/");
     public static final Prefix PREFIX_AMOUNT = new Prefix("m/");
+    public static final Prefix PREFIX_METADATA = new Prefix("d/");
```
Comment (chuahjiajie) [inline src/main/java/seedu/address/logic/parser/CliSyntax.java:16]: Why was "d/" chosen as the prefix?

File: src/main/java/seedu/address/model/person/Metadata.java [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+ */
+
+public class Metadata {
+    public static final String MESSAGE_CONSTRAINTS =
+            "Meta-data should only contain alphanumeric characters and spaces, and it should not be blank";
+
+    /*
+     * A valid metadata is any string that does not contain a `/`.
+     */
+    public static final String VALIDATION_REGEX = "[^/]+";
```
Comment (chuahjiajie) [inline src/main/java/seedu/address/model/person/Metadata.java:18]: Smart use of regex

File: src/main/java/seedu/address/model/util/SampleDataUtil.java [hunk 9]
```diff
@@ -23,22 +24,28 @@ public static Person[] getSamplePersons() {
...
+                new Metadata("I have food allergies: Egg and peanuts")),
             new Person(new Name("Charlotte Oliveiro"), new Phone("93210283"), new Email("charlotte@example.com"),
                 new Address("Blk 11 Ang Mo Kio Street 74, #11-04"),
-                getRoleSet("neighbours"), getCcaSet("NUS Cycling"), new Amount("0.0")),
+                getRoleSet("neighbours"), getCcaSet("NUS Cycling"), new Amount("0.0"),
+                new Metadata("I like to sleep on the floor.")),
             new Person(new Name("David Li"), new Phone("91031282"), new Email("lidavid@example.com"),
                 new Address("Blk 436 Serangoon Gardens Street 26, #16-43"),
-                getRoleSet("family"), getCcaSet("NUS Cycling"), new Amount("0.0")),
+                getRoleSet("family"), getCcaSet("NUS Cycling"), new Amount("0.0"),
```
Comment (chuahjiajie) [inline src/main/java/seedu/address/model/util/SampleDataUtil.java:39]: Good work updating all the sample data

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java [hunk 10]
```diff
@@ -30,18 +31,20 @@ class JsonAdaptedPerson {
     private final String phone;
     private final String email;
     private final String address;
+    private final String metadata;
```
Comment (chuahjiajie) [inline src/main/java/seedu/address/storage/JsonAdaptedPerson.java:34]: You sure only a string is enough, wouldn't it be better to make a JSONAdaptedMetaData?

**Review by chuahjiajie — COMMENTED** [review]
Overall looks good, just have a few questions. Thanks!

**Pull request merged**